crates/provider-wit-bindgen/src/lib.rs

```diff
@@ -1749,3 +1812,174 @@ fn convert_wit_type(t: &wit_parser::Type, cfg: &ProviderBindgenConfig) -> anyhow
         }
     }
 }
+
+/// Attempt to extract key and value types from a tree of tokens that is a witified map
+///
+/// For example, the following Rust type, submitted as a list of tokens would be parsed successfully:
+///
+/// ```rust,ignore
+/// Vec<(String, String)>
+/// ```
+fn extract_witified_map_types(input: &[TokenTree]) -> Option<(TokenStream, TokenStream)> {
+    match input {
+        [
+            TokenTree::Ident(vec_ty),
+            TokenTree::Punct(p1), // <
+            TokenTree::Group(g),
+            TokenTree::Punct(p2), // >
+        ] if *vec_ty == "Vec" && p1.to_string() == "<" && p2.to_string() == ">" => {
```

Comment:
Is it possible to have the tuple types themselves contain `<`, e.g. `Vec<(GenT<Foo>, GenT<Bar>)>`?

Reply:
Yeah it's possible -- the Group here should capture both of those.

That said, the comma-finding index right below works on simple cases of that, but it *could* get mixed up with a `Vec(GenT<Foo, Bar>, GenT<Baz, Bazr>)` -- *if* they don't show up as a `TokenTree::Group` (i.e. if there is more than one "top level" comma)

Reply:
Ahh, I was missing that `Group` would handle more complicated structures

Reply:
Oh just a word on this, it's entirely possible that I could get some tokens that *don't* behave like that... But I figure we can cross that bridge when we get there. I think it *is* possible that something comes in with some other combination of syntax possibilities that makes a bad comma show up.